fleet_management_api/api_impl/controllers/car_action.py

```
@@ -0,0 +1,210 @@
+            last_statuses[car_id] = state.action_status
+        # otherwise, the state is skipped, a.k.a. merged with the previous one
+    return invalid_state_transitions
+
+
+def get_last_action_states(*car_ids: int) -> list[CarActionState]:
+    """Get last action state for each car in car_ids."""
+    states = []
+    for car_id in car_ids:
+        db_model = _db_access.get(
+            _db_models.CarActionStateDBModel,
+            criteria={"car_id": lambda x: x == car_id},
```

Ruff (0.8.0)

Function definition does not bind loop variable `car_id`

(B023)